modules/core/output_modules.php

```diff
@@ -1225,11 +1225,10 @@ class Hm_Output_main_menu_content extends Hm_Output_Module {
     protected function output() {
         $res = '';
         $email = false;
-        $single = $this->get('single_server_mode');
         if (array_key_exists('email_folders', merge_folder_list_details($this->get('folder_sources', array())))) {
             $email = true;
         }
-        if (!$single) {
+        if (count($this->get('imap_servers', array()))  > 1) {
```

Comment:
Please check carefully what the All and Everything pages actually list. I think they aggregate more than the imap servers, so we need to be careful with the condition here.

Reply:
@Shadow243 I think jmap_servers, and feeds too as they also appear in everything should also be checked

Reply:
Oh copy that @josaphatim, let me check it out..

Reply:
@marclaporte @kroky 
It seems that EveryThing indeed contains several elements including feeds, emails etc...

This condition already applied therefore does not cover this scenario, however for the All menu under Email it works. which amounts to asking for clarification: will the task only cover All or should we also adapt Everything (apply a verification of all the elements concerned)??

Reply:
I'd suggest you check if all/everything actually displays more than one source - e.g. 2 imap accounts, 1 imap account and 1 rss feed, 1 imap account and 1 jmap account, etc. If it turns out there is only one source configured to display in All or Everything, then you can hide those links.

Reply:
Everything displays all configured sources. e.g. if we have 2 imap, 3 jmap, 1 feeds all datas will be visible in Everything.

However All under each menu(Email, Feeds) displays all sources each one based on parent. e.g: All under email displays All Email sources and All under Feeds displays all Feeds sources.

This is what confuses me a little, we will hide All(under Email and Feed) and let Everything as the main menu be displayed or also hide Everything(Main menu) if feeds and imap_servers(jmap_servers) only have one source configured each

According to my understanding, the first analysis is the correct one. An other idea ?

Reply:
In imap_servers array we have also jmap servers.

Somes changes added.

Reply:
We should be consistent. If there is only one email source, All menu item under email is hidden. Same for feeds and any other type of entity. As for 'Everything' - it will also be hidden if there is only one source of any type. In all other cases, it must be shown.

Reply:
That's what I did yesterday.

Reply:
Ok, just a minor comment and we are good.